### PR TITLE
chore: adjust ci workflow after release

### PR DESCRIPTION
**Pull request by mhaertwig**

### CODE REVIEW

File: .github/workflows/maintenance-changelog.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: cherry-pick changelog from release branch into default branch
 on:
   push:
     tags: ['*.*']
+    branches: ["*.x"]
```
Comment (jeripeierSBB) [inline .github/workflows/maintenance-changelog.yml:6]: why is this change needed?

Comment (mhaertwig) [reply .github/workflows/maintenance-changelog.yml:6]: Because otherwise this action tries to cherry-pick major releases (on main) into release branches (e.g. 15.x). However, we only want to allow it the other direction.

**Review by jeripeierSBB — APPROVED** [review]
LGTM
